### PR TITLE
fix: GEOT-324: mixin for buttons

### PR DESCRIPTION
**Pull request by ra8ga**

### CODE REVIEW

File: packages/app-andrew/src/components/commons/NotificationsSidebar/styles.scss [hunk 1]
```diff
@@ -1,33 +1,63 @@
...
-    span {
-      margin-left: 4px;
-      font-weight: 400;
-      font-size: 13px;
+    margin-bottom: 15px;
+
+    h2 {
+      display: flex;
+      align-items: center;
+      font-size: 18px;
```
Comment (mariusz-kabala) [inline packages/app-andrew/src/components/commons/NotificationsSidebar/styles.scss:23]: font-size is not a mixin

File: packages/app-andrew/src/components/commons/NotificationsSidebar/styles.scss [hunk 2]
```diff
@@ -1,33 +1,63 @@
...
+      @include font-weight(semiBold);
+      text-transform: uppercase;
+
+      span {
+        margin-left: 4px;
```
Comment (mariusz-kabala) [inline packages/app-andrew/src/components/commons/NotificationsSidebar/styles.scss:28]: wrong margin

File: packages/app-andrew/src/components/commons/NotificationsSidebar/styles.scss [hunk 3]
```diff
@@ -1,33 +1,63 @@
...
+    }
+
+    ul {
+      display: flex;
+      align-items: center;
+      margin-left: 10px;
+      cursor: pointer;
+
+      li {
+        padding: 3px 10px;
```
Comment (mariusz-kabala) [inline packages/app-andrew/src/components/commons/NotificationsSidebar/styles.scss:41]:
```suggestion
        padding: 5px 10px;
```

File: packages/app-andrew/src/components/commons/NotificationsSidebar/styles.scss [hunk 4]
```diff
@@ -1,33 +1,63 @@
...
+        border-radius: $borderRadius2;
+        margin-right: 10px;
+        font-size: 1.2em;
```
Comment (mariusz-kabala) [inline packages/app-andrew/src/components/commons/NotificationsSidebar/styles.scss:44]: why not mixin?

File: packages/app-andrew/src/components/group/FavsButton/index.tsx [hunk 5]
```diff
@@ -1,4 +1,5 @@
 import React, { FC, useState, useEffect } from 'react'
+import styles from './styles.scss'
```
Comment (mariusz-kabala) [inline packages/app-andrew/src/components/group/FavsButton/index.tsx:2]: stop putting styles at the top of import list, this should be the last item on that list. at the bottom

File: packages/app-andrew/src/components/group/FollowButton/styles.scss [hunk 6]
```diff
@@ -0,0 +1,17 @@
+@import '~@gtms/styles/scss/variables.scss';
+@import '~@gtms/styles/scss/mixins/fontWeight';
+
+.wrapper {
+  font-size: 12px;
```
Comment (mariusz-kabala) [inline packages/app-andrew/src/components/group/FollowButton/styles.scss:5]: use mixin

File: packages/app-andrew/src/components/group/GroupAvatar/index.tsx [hunk 7]
```diff
@@ -1,13 +1,13 @@
 import React, { FC, useState, useCallback } from 'react'
-import { Picture } from '@gtms/ui/Picture'
-import { UploadFile } from '@gtms/ui/UploadFile'
+import styles from './styles.scss'
```
Comment (mariusz-kabala) [inline packages/app-andrew/src/components/group/GroupAvatar/index.tsx:2]: fix those fucking imports, why did you touch it?

File: packages/app-andrew/src/components/login/SocialButtons/styles.scss [hunk 8]
```diff
@@ -1,40 +1,43 @@
...
-      text-align: center;
-      text-decoration: none;
-      opacity: 1 !important; // clear opacity coming from npm module
-      cursor: pointer;
+    &.facebook {
+      @include buttonDefault();
+      @include buttonSizeDefault();
+      @include buttonShapePrimary();
+      @include buttonColorsPrimary();
+      padding: 14px;
```
Comment (mariusz-kabala) [inline packages/app-andrew/src/components/login/SocialButtons/styles.scss:29]: padding: 15px

File: packages/app-andrew/src/pages/account/styles.scss [hunk 9]
```diff
@@ -1,54 +1,209 @@
...
+    ul {
+      display: flex;
+      align-items: center;
+      @include font-size(xdefault);
 
-    .avatar {
-      cursor: pointer;
+      li {
+        margin-left: 10px;
+        padding: 3px 15px;
```
Comment (mariusz-kabala) [inline packages/app-andrew/src/pages/account/styles.scss:42]:
```suggestion
        padding: 5px 15px;
```

File: packages/app-andrew/src/pages/group/styles.scss [hunk 10]
```diff
@@ -1,153 +1,144 @@
...
-        h2 {
-          font-weight: 400;
-          @include font-size('xdefault');
-          color: $theme7;
-        }
+    .descWrapper {
+      max-width: 400px;
+      min-width: 400px;
+      line-height: 1;
+      margin-left: 8px;
```
Comment (mariusz-kabala) [inline packages/app-andrew/src/pages/group/styles.scss:32]:
```suggestion
      margin-left: 10px;
```

File: packages/app-andrew/src/pages/group/styles.scss [hunk 11]
```diff
@@ -1,153 +1,144 @@
...
+    display: flex;
+    overflow: hidden;
+
+    .search {
       display: flex;
-      flex-direction: column;
       width: 100%;
-      margin: 15px 0;
+      border-radius: $borderRadius1;
+      height: 38px;
```
Comment (mariusz-kabala) [inline packages/app-andrew/src/pages/group/styles.scss:67]: why 38px? 40px

File: packages/app-andrew/src/pages/login/styles.scss [hunk 12]
```diff
@@ -2,56 +2,145 @@
...
+      }
     }
   }
 }
+
+.tagsGrid {
+  position: relative;
+  padding: 0 40px;
+  // overlap on main photo
+  // padding-top: 50px;
```
Comment (mariusz-kabala) [inline packages/app-andrew/src/pages/login/styles.scss:123]: remove comments

File: packages/app-andrew/src/pages/my-groups/styles.scss [hunk 13]
```diff
@@ -1,30 +1,80 @@
...
+    }
+
+    .items {
+      display: flex;
+
+      .item {
+        display: flex;
+        align-items: center;
+        margin-right: 10px;
+        padding: 6px 15px;
```
Comment (mariusz-kabala) [inline packages/app-andrew/src/pages/my-groups/styles.scss:38]:
```suggestion
        padding: 5px 15px;
```

File: packages/app-andrew/src/pages/registration/styles.scss [hunk 14]
```diff
@@ -1,50 +1,141 @@
...
+        grid-template-columns: 1fr 1fr;
+      }
     }
   }
 }
+
+.tagsGrid {
+  position: relative;
+  padding: 0 40px;
+  // overlap on main photo
```
Comment (mariusz-kabala) [inline packages/app-andrew/src/pages/registration/styles.scss:117]: remove comments

File: packages/styles/scss/mixins/buttons.scss [hunk 15]
```diff
@@ -8,7 +9,7 @@
   outline: none;
   justify-content: center;
   padding: 0px;
-  text-align: left; // for <a> elements
+  text-align: center; // for <a> elements
```
Comment (mariusz-kabala) [inline packages/styles/scss/mixins/buttons.scss:12]: remove those comments, does not make sense

File: packages/ui/Button/index.tsx [hunk 16]
```diff
@@ -5,7 +5,7 @@ export const Button: FC<{
   children: ReactNode
   testid?: string
   disabled?: boolean
-  onClick?: () => unknown
+  onClick?: (x: any) => unknown
```
Comment (mariusz-kabala) [inline packages/ui/Button/index.tsx:8]: wtf is this any here

File: packages/ui/Navigation/styles.scss [hunk 17]
```diff
@@ -27,22 +27,26 @@
...
         border-radius: $borderRadius1;
-        background: $theme4;
+        background: rgba($theme4, 0.2);
+        padding: 2px;
 
         a {
           position: relative;
           display: block;
-          padding: 11px;
+          padding: 9px;
```
Comment (mariusz-kabala) [inline packages/ui/Navigation/styles.scss:36]: whyy 9px

File: packages/ui/NavigationPage/styles.scss [hunk 18]
```diff
@@ -3,100 +3,60 @@
...
+        display: block;
+        color: $theme7;
+      }
 
-        span {
-          margin-left: 10px;
-          color: $theme7;
-        }
+      span {
+        margin-left: 4px;
```
Comment (mariusz-kabala) [inline packages/ui/NavigationPage/styles.scss:50]:
```suggestion
        margin-left: 5px;
```

File: packages/ui/Overlay/index.tsx [hunk 19]
```diff
@@ -13,6 +13,6 @@ export const Overlay: FC<{
     })}
     data-testid="overlay"
     onClick={onClick}
-    style={{ opacity: opacity ? opacity : 0.75 }}
+    style={{ opacity: opacity ? opacity : 0.8 }}
```
Comment (mariusz-kabala) [inline packages/ui/Overlay/index.tsx:16]: use a class name here, not inline styles

File: packages/ui/PostSingle/styles.scss [hunk 20]
```diff
@@ -44,21 +44,25 @@
...
+      .date {
+        @include font-weight(semiNormal);
+      }
 
-      // @include font-weight('bold');
-      &:hover {
-        color: $theme4;
+      span {
+        margin-left: 10px;
+        // @include font-weight('bold');
```
Comment (mariusz-kabala) [inline packages/ui/PostSingle/styles.scss:54]: remove comment

File: packages/ui/Spinner/index.tsx [hunk 21]
```diff
@@ -4,8 +4,12 @@ import styles from './styles.scss'
 
 export const Spinner: FC<{
   additionalStyles?: string
-}> = ({ additionalStyles }) => (
-  <div className={cx(styles.wrapper, additionalStyles)} data-testid="spinner">
+  testid?: string
+}> = ({ additionalStyles, testid }) => (
+  <div
+    className={cx(styles.wrapper, additionalStyles)}
+    data-testid={testid ?? 'spinner'}
```
Comment (mariusz-kabala) [inline packages/ui/Spinner/index.tsx:11]: why do you need a custom testid?

File: packages/app-andrew/src/components/group/FollowButton/styles.scss (hunk 6 above)
Comment (ra8ga) [reply packages/app-andrew/src/components/group/FollowButton/styles.scss:5]: fixed

File: packages/app-andrew/src/components/commons/NotificationsSidebar/styles.scss (hunk 1 above)
Comment (ra8ga) [reply packages/app-andrew/src/components/commons/NotificationsSidebar/styles.scss:23]: fixed

File: packages/app-andrew/src/components/commons/NotificationsSidebar/styles.scss (hunk 2 above)
Comment (ra8ga) [reply packages/app-andrew/src/components/commons/NotificationsSidebar/styles.scss:28]: fixed

File: packages/app-andrew/src/components/commons/NotificationsSidebar/styles.scss (hunk 4 above)
Comment (ra8ga) [reply packages/app-andrew/src/components/commons/NotificationsSidebar/styles.scss:44]: fixed

File: packages/app-andrew/src/components/group/FavsButton/index.tsx (hunk 5 above)
Comment (ra8ga) [reply packages/app-andrew/src/components/group/FavsButton/index.tsx:2]: fixed

File: packages/app-andrew/src/components/group/GroupAvatar/index.tsx (hunk 7 above)
Comment (ra8ga) [reply packages/app-andrew/src/components/group/GroupAvatar/index.tsx:2]: fixed

File: packages/app-andrew/src/components/login/SocialButtons/styles.scss (hunk 8 above)
Comment (ra8ga) [reply packages/app-andrew/src/components/login/SocialButtons/styles.scss:29]: fixed

File: packages/app-andrew/src/pages/account/styles.scss (hunk 9 above)
Comment (ra8ga) [reply packages/app-andrew/src/pages/account/styles.scss:42]: fixed

File: packages/app-andrew/src/pages/group/styles.scss (hunk 10 above)
Comment (ra8ga) [reply packages/app-andrew/src/pages/group/styles.scss:32]: fixed

File: packages/app-andrew/src/pages/my-groups/styles.scss (hunk 13 above)
Comment (ra8ga) [reply packages/app-andrew/src/pages/my-groups/styles.scss:38]: fixed

File: packages/app-andrew/src/pages/registration/styles.scss (hunk 14 above)
Comment (ra8ga) [reply packages/app-andrew/src/pages/registration/styles.scss:117]: fixed

File: packages/styles/scss/mixins/buttons.scss (hunk 15 above)
Comment (ra8ga) [reply packages/styles/scss/mixins/buttons.scss:12]: fixed

File: packages/ui/Button/index.tsx (hunk 16 above)
Comment (ra8ga) [reply packages/ui/Button/index.tsx:8]: fixed

File: packages/ui/Navigation/styles.scss (hunk 17 above)
Comment (ra8ga) [reply packages/ui/Navigation/styles.scss:36]: fixed

File: packages/ui/NavigationPage/styles.scss (hunk 18 above)
Comment (ra8ga) [reply packages/ui/NavigationPage/styles.scss:50]: fixed

File: packages/app-andrew/src/pages/login/styles.scss (hunk 12 above)
Comment (ra8ga) [reply packages/app-andrew/src/pages/login/styles.scss:123]: fixed

File: packages/ui/Overlay/index.tsx (hunk 19 above)
Comment (ra8ga) [reply packages/ui/Overlay/index.tsx:16]: question here

File: packages/ui/PostSingle/styles.scss (hunk 20 above)
Comment (ra8ga) [reply packages/ui/PostSingle/styles.scss:54]: fixed

File: packages/ui/Spinner/index.tsx (hunk 21 above)
Comment (ra8ga) [reply packages/ui/Spinner/index.tsx:11]: removed

File: packages/ui/Overlay/index.tsx (hunk 19 above)
Comment (ra8ga) [reply packages/ui/Overlay/index.tsx:16]: fixed

File: packages/app-andrew/src/pages/group/styles.scss (hunk 11 above)
Comment (ra8ga) [reply packages/app-andrew/src/pages/group/styles.scss:67]: fixed

**Pull request merged**